src/lib.rs

```diff
@@ -4,135 +4,159 @@ use core::sync::atomic;
 use core::marker::PhantomData;
 
 use std::boxed::Box;
-use std::mem;
 use std::sync::atomic::AtomicUsize;
 
 use std::sync::atomic::Ordering;
 
-pub struct Mlsp<T: ?Sized> {
-    local_count: *mut usize,
-    atomic_count: *mut AtomicUsize,
-    
-    ptr: *mut T,
-
-    phantom: PhantomData<T>
+struct Packet<T> {
+    atomic_count: AtomicUsize,
+    value: T
 }
 
-pub struct MlspPackage<T: ?Sized> {
-    atomic_count: *mut AtomicUsize,
-    
-    ptr: *mut T,
-
-    phantom: PhantomData<T>
+pub struct Mlsp<T> {
+    phantom: PhantomData<T>,
```

Comment:
This `PhantomData` is not needed anymore, unless the `Mlsp` logically owns `T`.

Reply:
Thanks